src/index.js

```diff
@@ -24,9 +34,48 @@ const products = [
 ]
 // Target the ul element in index.html and render the above list
 
+const productList = document.querySelector(".product-list");
+products.map(product => {
+  const productListItem = document.createElement("li");
+  productListItem.textContent = product.name;
+  productList.appendChild(productListItem);  
+})
+
 // 7. Target the button in index.html, attach an event listener to it. Every time we click
 // on the button, it will toggle the product list. Show => hide, hide => show
 
+const toggleButton = document.querySelector(".toggle");
+productList.style.display = "none";
+toggleButton.textContent = "Show";
+toggleButton.addEventListener("click", () => {
+  if(productList.style.display === "block"){
+    productList.style.display = "none";
+    toggleButton.textContent = "Show";
+  } else {
+    productList.style.display = "block";
+    toggleButton.textContent = "Hide";
+  }
+})
+
 // 8. Target the input element and attach an event listener to it.
 // Once we click search button, you need to use the text input to search for the relevant product name from the list above
 // and show it in the <span> tag
+
+const searchInput = document.querySelector(".search-input");
+const searchButton = document.querySelector(".search-button");
+const result = document.querySelector(".result");
+
+searchButton.addEventListener("click", (e) => {
```

Comment:
The more standard way would be to add event listener to the form itself, and listen for `submit` event.